src/client/java/daos-java-api/src/main/native/com_intel_daos_client_DaosFsClient.c

```diff
@@ -0,0 +1,911 @@
+			char *msg = (char *)malloc(strlen(tmp) + strlen(parent_path));
+			sprintf(msg, tmp, parent_path);
+			throw_exception(env, msg, rc);
+			goto out;
+		}
+	}
+
+	rc = dfs_open(dfs, parent, file_name, S_IFREG | mode, O_CREAT | accessFlags, objectId, chunkSize, NULL, &file);
+	if (rc) {
+		char *tmp = "Failed to create new file (%s) under directory (%s)";
+		char *msg = (char *)malloc(strlen(tmp) + strlen(file_name) + strlen(parent_path));
+		sprintf(msg, tmp, file_name, parent_path);
+		throw_exception(env, msg, rc);
+	}
+
+out:
+	(*env)->ReleaseStringUTFChars(env, parentPath, parent_path);
+	(*env)->ReleaseStringUTFChars(env, name, file_name);
+	if (parentError) free(parentError);
+	if (parent) dfs_release(parent);
+	return *(jlong*)&file;;
+}
+
+JNIEXPORT jboolean JNICALL Java_com_intel_daos_client_DaosFsClient_delete
+  (JNIEnv *env, jobject object, jlong dfsPtr, jstring parentPath, jstring name, jboolean force){
+	dfs_t *dfs = *(dfs_t**)&dfsPtr;
+	const char *parent_path = (*env)->GetStringUTFChars(env, parentPath, NULL);
+	const char *file_name = (*env)->GetStringUTFChars(env, name, NULL);
+	dfs_obj_t *parent = NULL;
+	mode_t tmp_mode;
+	int rc, ret;
+	if ((strlen(parent_path) > 0) && (strcmp(parent_path, "/") != 0)){
+	    rc = dfs_lookup(dfs, parent_path, O_RDWR, &parent, &tmp_mode, NULL);
+	    if(rc){
+	    	printf("Failed to open parent dir, %s, when delete, rc: %d, error msg: %s\n", parent_path, rc, strerror(rc));
+	    	ret = 0;
+	    	goto out;
+	    }
+	}
+	rc = dfs_remove(dfs, parent, file_name, force, NULL);
+	if (rc) {
+		printf("Failed to delete %s from %s, rc: %d, error msg: %s\n", file_name, parent_path, rc, strerror(rc));
+		ret = 0;
+		goto out;
+	}
+	ret = 1;
+out:
+	(*env)->ReleaseStringUTFChars(env, parentPath, parent_path);
+	(*env)->ReleaseStringUTFChars(env, name, file_name);
+	if (parent) dfs_release(parent);
+	return ret;
+}
+
+/**
+ * TODO: bufferAddress is to be considered to pull StatAttributes
+ */
+JNIEXPORT jlong JNICALL Java_com_intel_daos_client_DaosFsClient_dfsLookup__JJLjava_lang_String_2IJ
+  (JNIEnv *env, jobject client, jlong dfsPtr, jlong parentObjId, jstring name, jint flags, jlong bufferAddress){
+	dfs_t *dfs = *(dfs_t**)&dfsPtr;
+	dfs_obj_t *parent = *(dfs_obj_t**)&parentObjId;
+	dfs_obj_t *file;
+	mode_t tmp_mode;
+	const char *file_name = (*env)->GetStringUTFChars(env, name, NULL);
+	int rc = dfs_lookup_rel(dfs, parent, file_name, flags, &file, &tmp_mode, NULL);
+	if (rc) {
+		char *tmp = "Failed to open file (%s) under parent with flags (%d)";
+		char *msg = (char *)malloc(strlen(tmp) + strlen(file_name) + 4);
+		sprintf(msg, tmp, file_name, flags);
+		throw_exception(env, msg, rc);
+		file = NULL;
+	}
+	(*env)->ReleaseStringUTFChars(env, name, file_name);
+	return *(jlong*)&file;
+}
+
+/**
+ * TODO: bufferAddress is to be considered to pull StatAttributes
+ */
+JNIEXPORT jlong JNICALL Java_com_intel_daos_client_DaosFsClient_dfsLookup__JLjava_lang_String_2IJ
```

Comment:
(style)  line over 80 characters
(style)  arguments for function declarations should follow identifier